ethcore/src/types/transaction.rs

```diff
@@ -135,26 +140,26 @@ impl From<ethjson::transaction::Transaction> for SignedTransaction {
 
 impl Transaction {
 	/// The message hash of the transaction.
-	pub fn hash(&self) -> H256 {
+	pub fn hash(&self, network_id: Option<u8>) -> H256 {
 		let mut stream = RlpStream::new();
-		self.rlp_append_unsigned_transaction(&mut stream);
+		self.rlp_append_unsigned_transaction(&mut stream, network_id);
 		stream.out().sha3()
 	}
 
 	/// Signs the transaction as coming from `sender`.
-	pub fn sign(self, secret: &Secret) -> SignedTransaction {
-		let sig = ::ethkey::sign(secret, &self.hash())
+	pub fn sign(self, secret: &Secret, network_id: Option<u8>) -> SignedTransaction {
```

Comment:
standard would be a misnomer - could be `sign_without_network_id`, but then i think `sign(..., None)` says mostly the same without the verbosity and additional `fn`.